src/js/system/RoutingService.js

```diff
@@ -0,0 +1,113 @@
+import {Route} from 'react-router';
+
+const queue = {};
+
+class Tab {
+  constructor(path, component) {
+    this.path = path;
+    this.component = component;
+  }
+
+  getPath() {
+    return this.path;
+  }
+
+  getComponent() {
+    return this.component;
+  }
+}
+
+class Page {
+  constructor(path, component) {
+    this.path = path;
+    this.component = component;
+    this.tabs = [];
+  }
+
+  getPath() {
+    return this.path;
+  }
+
+  getComponent() {
+    return this.component;
+  }
+
+  getTabs() {
+    return this.tabs;
+  }
+
+  addTab(path, component) {
+    const tab = new Tab(path, component);
+    this.tabs.push(tab);
+  }
+}
+
+const RoutingService = {
+  registerPage(path, component) {
+    if (!component) {
+      throw new Error(`Please provide a component for the new page ${path}!`);
+    }
+    if (queue[path]) {
+      throw new Error(`Page with the path ${path} is already registered in the queue!`);
+    }
+
+    queue[path] = new Page(path, component);
+
+    return queue[path];
+  },
+
+  findPage(path) {
+    let page = queue[path];
+    if (!page) {
+      queue[path] = new Page(path);
+    }
+
+    return page;
+  },
+
+  render(routes) {
```

Comment:
We need to make sure that we can run this method several times without overwriting page routes and attaching the same tabs over and over again.